cmd/csi-node-driver-registrar/util_linux.go

```diff
@@ -0,0 +1,27 @@
+// +build linux
+
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"golang.org/x/sys/unix"
+)
+
+func umask(mask int) (int, error) {
+	return unix.Umask(mask), nil
```

Comment:
what is purpose of error here?

Reply:
Will remove.

Reply:
We need the wrapper function to able to allow platform dependent building (and skip this for Windows). The error is necessary in the Windows wrapper to clearly indicate that `umask` is not implemented on Windows. So we need the error returned for the _linux case as well.

Reply:
With this wrapper function, do you still need to check OS in the code when using umask?

	if runtime.GOOS == "linux" {
		// Default to only user accessible socket, caller can open up later if desired
		oldmask, _ = umask(0077)
	}